spring-data-geode/src/test/java/org/springframework/data/gemfire/function/execution/onservers/FunctionsReturnResultsFromAllServersIntegrationTests.java

```diff
@@ -0,0 +1,172 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.springframework.data.gemfire.function.execution.onservers;
+
+import org.apache.geode.StatisticDescriptor;
+import org.apache.geode.Statistics;
+import org.apache.geode.StatisticsType;
+import org.apache.geode.cache.Cache;
+import org.apache.geode.cache.CacheFactory;
+import org.apache.geode.cache.execute.Function;
+import org.apache.geode.cache.execute.FunctionContext;
+import org.apache.geode.cache.execute.FunctionService;
+import org.apache.geode.cache.server.CacheServer;
+import org.apache.geode.distributed.internal.InternalDistributedSystem;
+import org.apache.geode.internal.statistics.StatisticsManager;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.data.gemfire.config.annotation.ClientCacheApplication;
+import org.springframework.data.gemfire.function.config.EnableGemfireFunctionExecutions;
+import org.springframework.data.gemfire.process.ProcessWrapper;
+import org.springframework.data.gemfire.test.support.ClientServerIntegrationTestsSupport;
+import org.springframework.data.gemfire.transaction.config.EnableGemfireCacheTransactions;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * @author Patrick Johnson
+ */
+@SuppressWarnings("unused")
+@RunWith(SpringRunner.class)
+@ContextConfiguration(classes = FunctionsReturnResultsFromAllServersIntegrationTests.Config.class)
+public class FunctionsReturnResultsFromAllServersIntegrationTests extends ClientServerIntegrationTestsSupport {
+
+	private static final int PORT_1 = 40407;
+	private static final int PORT_2 = 40403;
+
+	private static ProcessWrapper gemfireServer1;
+	private static ProcessWrapper gemfireServer2;
+
+	@Autowired
+	private AllServersAdminFunctions allServersAdminFunctions;
+
+	@Autowired
+	private SingleServerAdminFunctions singleServerAdminFunctions;
+
+	@BeforeClass
+	public static void startGemFireServer() throws Exception {
+
+		gemfireServer1 = run(MetricsFunctionServerProcess.class,
+				String.format("-D%s=%d", GEMFIRE_CACHE_SERVER_PORT_PROPERTY, PORT_1));
+
+		waitForServerToStart(DEFAULT_HOSTNAME, PORT_1);
+
+		gemfireServer2 = run(MetricsFunctionServerProcess.class,
+				String.format("-D%s=%d", GEMFIRE_CACHE_SERVER_PORT_PROPERTY, PORT_2));
+
+		waitForServerToStart(DEFAULT_HOSTNAME, PORT_2);
+	}
+
+	@ClientCacheApplication(servers = {@ClientCacheApplication.Server(port = PORT_1), @ClientCacheApplication.Server(port = PORT_2)})
```

Comment:
The hardcode server Pool configuration would not be necessary using the dynamic approach I referred to above.

For [instance](https://github.com/spring-projects/spring-data-geode/blob/master/spring-data-geode/src/test/java/org/springframework/data/gemfire/config/annotation/EnableSslConfigurationIntegrationTests.java#L94).

Reply:
No longer hard-coding